mcpy/mcsimulation.py

```diff
@@ -42,54 +42,62 @@ class MCSimulation(object):
         Determines when to log data to std_out, default 10000.
     '''
 
-    def __init__(self, tune_integrators=True, frequency=10000):
+    def __init__(self, beta=1, tune_integrators=True, frequency=10000):
         self._tuning = tune_integrators
+        self.beta = beta
         self.frequency = frequency
         self.step = 0
         self.steps_accepted = []
+        self._integrators = []
         self._log_index = CounterIndex()
 
     def run(self, steps):
         self.check_state()
         self._initialize_state(steps)
         for i in range(steps):
-            for i, integrator in enumerate(self.integrators):
-                self.steps += 1
+            for i, integrator in enumerate(self._integrators):
+                self.step += 1
                 acceptance_rate = self.steps_accepted[i] / self.step
-                delta_e, accepted = integrator(self.particles,
+                accepted, delta_e = integrator(self.particles,
                                                self.box,
-                                               acceptance_rate,
-                                               tune=self.tune)
+                                               self.beta,
+                                               tune_displacement=self.tune,
+                                               acc_rate=acceptance_rate)
                 if accepted:
                     self.steps_accepted[i] += 1
                     self.energy += delta_e
-                if self.steps % self.frequency == 0:
+                if self.step % self.frequency == 0:
                     self.print_log()
                     self._update_log()
+        self.print_log()
+        self._update_log()
 
     def run_upto(self, step):
         if self.step >= self.step:
             return None
         self.run(step - self.step)
 
     def _update_log(self):
-        index = self._log_index
+        index = self._log_index()
```

Comment:
i m not sure but, _log_index seems like a variable self._log_index, I m not sure if the '()' needs to be removed